api/envoy/extensions/filters/http/ext_proc/v3alpha/processing_mode.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.ext_proc.v3alpha;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.ext_proc.v3alpha";
+option java_outer_classname = "ProcessingModeProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: External Processing Filter]
+// External Processing Filter Processing Mode
+// [#extension: envoy.filters.http.ext_proc]
+
+// This configuration describes how an external processing filter processes the HTTP
+// request or response, including how requests and responses are
+// sent to the remote server.
+
+// [#next-free-field: 9]
+message ProcessingMode {
+  enum SyncMode {
+    // The default SyncMode depends on which part of the message is being set.
+    SYNC_DEFAULT = 0;
+
+    // Send the header and wait for a response before continuing with
+    // the filter chain.
+    SYNC = 1;
+
+    // Send the header but to not expect a response, continuing immediately
+    // with the filter chain.
+    ASYNC = 2;
+
+    // Do not send the header.
+    SKIP = 3;
+  }
+
+  enum BodyMode {
+    // The default SyncMode depends on which part of the message is being set.
+    BODY_DEFAULT = 0;
+
+    // Send each chunk of the body separately to the server.
```

Comment:
I'd be inclined to call this streaming.  Chunked could be confused with HTTP/1.1 chunks and we're not actually sending a chunk at a time.

Reply:
Makes sense -- let's not use the "chunk" term.